daemon/graphdriver/aufs/aufs.go

```diff
@@ -272,9 +273,27 @@ func (a *Driver) Remove(id string) error {
 	if !exists {
 		mountpoint = a.getMountpoint(id)
 	}
-	if err := a.unmount(mountpoint); err != nil {
-		// no need to return here, we can still try to remove since the `Rename` will fail below if still mounted
-		logrus.Debugf("aufs: error while unmounting %s: %v", mountpoint, err)
+
+	var retries int
+	for {
+		if err := a.unmount(mountpoint); err != nil {
+			if err != syscall.EBUSY {
```

Comment:
unmount() considered not mounted paths as an error. I can call unmount only if path is mounted, else we can proceed to rename/remove.